### PR TITLE
fix: BJData size calculation overflow

### PR DESCRIPTION
**Pull request by thevilledev**
Adds pre-multiplication overflow detection to catch cases where dimension products would exceed size_t max. The previous check only detected when overflow resulted in exactly 0 or SIZE_MAX, missing other cases.

Retains the original post-multiplication check for backward compatibility. Adds tests verifying overflow detection with dimensions (2^32+1)×(2^32), which previously overflowed silently to 2^32.

This prevents custom SAX handlers from receiving incorrect array sizes that could lead to buffer overflows.

- [x] The changes are described in detail, both the what and why.
- [ ] If applicable, an [existing issue](https://github.com/nlohmann/json/issues) is referenced.
- [x] The [Code coverage](https://coveralls.io/github/nlohmann/json) remained at 100%. A test case for every new line of code.
- [ ] If applicable, the [documentation](https://json.nlohmann.me) is updated.
- [x] The source code is amalgamated by running `make amalgamate`.

### CODE REVIEW

File: tests/src/unit-bjdata.cpp [hunk 1]
```diff
@@ -2815,6 +2815,126 @@ TEST_CASE("BJData")
...
+                    0x4D,                                           // 'M' (uint64)
+                    0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, // 2^32 (4294967296) as little-endian
+                    0x5D                                            // ']' end dimensions
+                    // No data - we don't need it for this test, we just want to hit the overflow check
+                };
+
+                // Test with overflow dimensions using SAX parser
+                {
+                    SimpleOverflowSaxHandler handler;
+                    auto result = json::sax_parse(bjdata_payload, &handler,
```
Comment (nlohmann) [inline tests/src/unit-bjdata.cpp]: This variable is not used which is a warning/error in the used compilers.

Comment (thevilledev) [reply tests/src/unit-bjdata.cpp]: Yup, will push a fix soon!

Comment (nlohmann) [reply tests/src/unit-bjdata.cpp]: Thanks! There are also some warnings/errors unrelated to your changes which will be fixed with #4763.

**Comment by nlohmann** [comment]
Please rebase to the latest `develop` branch to fix the GCC-related errors.

**Review by nlohmann — APPROVED** [review]
Looks good to me.

**Comment by nlohmann** [comment]
Thanks!
